packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -51,7 +53,10 @@ const FormWithRedirect: FC<FormWithRedirectProps> = ({
     initialValues,
     initialValuesEqual,
     keepDirtyOnReinitialize = true,
-    mutators = arrayMutators as any, // FIXME see https://github.com/final-form/react-final-form/issues/704 and https://github.com/microsoft/TypeScript/issues/35771
+    mutators = {
```

Comment:
I'd create a `defaultMutators` outside of the component, to avoid that the `mutators` value change at each render